src/hybrid_reader.c

```diff
@@ -127,9 +127,16 @@ void computeDistances(HybridIterator *hr) {
   RSIndexResult *cur_res = hr->base.current;
   RSIndexResult *cur_child_res;  // This will use the memory of hr->child->current.
   RSIndexResult *cur_vec_res = NewDistanceResult();
+  void *qvector = hr->query.vector;
+
+  if (hr->indexMetric == VecSimMetric_Cosine) {
+    qvector = rm_malloc(hr->dimension * VecSimType_sizeof(hr->vecType));
+    memcpy(qvector, hr->query.vector, hr->dimension * VecSimType_sizeof(hr->vecType));
```

Comment:
Why do you need to copy? can't you just normalise `hr->query.vector` in place?

Reply:
We previously decided that the blob in the PARAMS part is a read only. Assume that when we support multiple VSS in one query, client can pass the same blob for L2 and Cosine indexes